frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -622,6 +642,35 @@ const getFilters = (t: TFunction): RowFilter<NodeRowItem>[] => [
       return input.selected?.some((r) => nodeRoles.includes(r));
     },
   },
+  {
+    filterGroupName: t('console-app~CPU Architecture'),
+    type: 'cpu-arch',
+    reducer: (obj) => (isCSRResource(obj) ? 'amd64' : getNodeCpuArch(obj)),
+    items: [
+      { id: 'amd64', title: t('console-app~amd64') },
+      { id: 'ppc64le', title: t('console-app~ppc64le') },
+      { id: 'arm64', title: t('console-app~arm64') },
+      { id: 's390x', title: t('console-app~s390x') },
+    ],
+    filter: (input, obj) => {
+      if (!input.selected?.length) {
+        return true;
+      }
+      if (isCSRResource(obj)) {
+        return input.selected?.includes('amd64');
+      }
+      if (isCSRResource(obj)) {
```

Comment:
This logic looks suspicious to me. Wouldn't we always return a value just above if `isCSRResource(obj)` is true?

Reply:
Yes , Modified to a single line.